core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -19,20 +19,45 @@ package org.http4s
 import cats.Monoid
 import cats.syntax.all._
 import cats.~>
+import fs2.Pure
+import fs2.Stream
 
-final case class Entity[+F[_]](body: EntityBody[F], length: Option[Long] = None) {
-  def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G] = Entity(body.translate(fk), length)
+sealed trait Entity[+F[_]] {
+  def body: EntityBody[F]
+  def length: Option[Long]
+  def ++[F1[x] >: F[x]](that: Entity[F1]): Entity[F1]
+  def translate[F1[x] >: F[x], G[_]](fk: F1 ~> G): Entity[G]
 }
 
 object Entity {
+  def apply[F[_]](body: EntityBody[F], length: Option[Long] = None): Entity[F] =
+    Default(body, length)
+
+  // The type parameter aids type inference in Message constructors. This should be unnecessary when Messages are covariant.
+  def empty[F[_]]: Entity[F] = Empty
+
+  final case class Default[+F[_]](body: EntityBody[F], length: Option[Long]) extends Entity[F] {
```

Comment:
Default is a placeholder for cases that haven't yet been specialized.

Reply:
It's hard-coded to `Some(0L)` in the empty case.

Reply:
Never mind, I misread what line of code you were commenting on. I guess this is superseded by the discussion below.